shell/browser/ui/file_dialog_linux.cc

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2024 Microsoft, Gmbh.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include <memory>
```

Comment:
Can you check if `shell_dialog_linux::Initialize()` is called during or soon after Electron initialization? If not, it should be called sometime early so that portal availability can be verified. This would avoid delays while opening the first dialog on systems that don't support portals.

Reply:
@aiddya yes! `shell_dialog_linux::Initialize()` is called by https://github.com/electron/electron/blob/b428315c6dd9aea97611ab3ef98145b870ef33ae/shell/browser/electron_browser_main_parts.cc#L397